packages/dropcaps.lua

```diff
@@ -0,0 +1,90 @@
+SILE.require("packages/rebox")
+SILE.require("packages/raiselower")
+
+local shapeHbox = function (options, content)
+  -- Clear irrelevant values before passing to font
+  options.lines, options.join, options.raise, options.shift, options.color = nil, nil, nil, nil, nil
+  SILE.call("noindent")
+  local hbox = SILE.call("hbox", {}, function ()
+    SILE.call("font", options, content)
+  end)
+  table.remove(SILE.typesetter.state.nodes)
+  return hbox
+end
+
+-- This implementation relies on the hangafter and hangindent features of Knuth's line-breaking algorithm.
+-- These features in core line breaking algorithm supply the blank space in the paragraph shape but don't fill it with anything.
+SILE.registerCommand("dropcap", function (options, content)
+  local lines = SU.cast("integer", options.lines or 3)
+  local join = SU.boolean(options.join, false)
+  local standoff = SU.cast("measurement", options.standoff or "1spc")
+  local raise = SU.cast("measurement", options.raise or 0)
+  local shift = SU.cast("measurement", options.shift or 0)
+  local size = SU.cast("measurement or nil", options.size or nil)
+  local color = options.color
+  options.size = nil -- we need to measure the "would have been" size before using this
+
+  if color then SILE.require("packages/color") end
+
+  -- We want the drop cap to spanning N lines is N-1 baselineskip plus the height of the first line.
+  -- We Define the height of the first line based on measuring the size the initial would have been.
```

Comment:
(Cosmetic) The comments there have some typos and might be clear. Suggested: 

> We want the drop cap to **span over** N lines, **that** is N-1 baselineskip plus the height of the first line.

> We **d**efine the height (...)

Reply:
Thanks, fixed.